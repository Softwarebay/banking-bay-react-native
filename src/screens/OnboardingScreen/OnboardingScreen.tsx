import { useNavigation } from '@react-navigation/native';
import { useState } from 'react';

import { OnboardingTemplate } from 'components/templates';
import type { ImageProps } from 'components/atoms';
import type { NavigationProp } from 'navigation';

interface OnboardingSlide {
	id: number;
	title: string;
	description: string;
	img: ImageProps['name'];
}

const onboardingSlides: OnboardingSlide[] = [
	{
		id: 1,
		title: 'Welcome to BankingBay app!',
		description:
			'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
		img: 'onboarding1',
	},
	{
		id: 2,
		title: 'Get a new card in a few clicks!',
		description:
			'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
		img: 'onboarding2',
	},
	{
		id: 3,
		title: 'Easy payments all over the world!',
		description:
			'Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
		img: 'onboarding3',
	},
];

export const OnboardingScreen = () => {
	const [currentSlideIndex, setCurrentSlideIndex] = useState(0);
	const navigation = useNavigation<NavigationProp>();

	const onPressOnboardingBtn = () => {
		navigation.navigate('SignIn');
	};

	const onScroll = (index: number) => {
		setCurrentSlideIndex(index);
	};

	return (
		<OnboardingTemplate
			currentSlideIndex={currentSlideIndex}
			onboardingSlides={onboardingSlides}
			onPressOnboardingBtn={onPressOnboardingBtn}
			onScroll={onScroll}
		/>
	);
};
