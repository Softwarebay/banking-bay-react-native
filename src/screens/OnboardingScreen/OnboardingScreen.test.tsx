import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { OnboardingScreen } from './OnboardingScreen';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
	useNavigation: () => ({ navigate: mockNavigate }),
}));

describe('OnboardingScreen', () => {
	it('should match snapshot', () => {
		const tree = testRenderer.create(<OnboardingScreen />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should navigate to SignInScreen after pressing on 'Get Started' button", () => {
		render(<OnboardingScreen />);

		const btn = screen.getByTestId('onboarding-btn-test-id');

		fireEvent.press(btn);

		expect(mockNavigate).toHaveBeenCalledWith('SignIn');
	});
});
