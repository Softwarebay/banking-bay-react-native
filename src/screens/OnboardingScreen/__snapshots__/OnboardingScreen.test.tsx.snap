// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingScreen should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
        "left",
        "right",
        "top",
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": undefined,
          },
          Object {
            "aspectRatio": 0.7025,
          },
          Object {
            "top": 50,
            "width": "95%",
          },
        ]
      }
      testID="onboarding1-img-test-id"
    />
  </RNCSafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "#FAFCFF",
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "flex": 1,
        "marginHorizontal": 25,
        "paddingBottom": 0,
        "paddingHorizontal": 25,
        "paddingTop": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <SvgMock
        color="#1B1D4D"
        height={40}
        testID="logo-icon-test-id"
        width={40}
      />
      <RCTScrollView
        data={
          Array [
            Object {
              "description": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "id": 1,
              "img": "onboarding1",
              "title": "Welcome to BankingBay app!",
            },
            Object {
              "description": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "id": 2,
              "img": "onboarding2",
              "title": "Get a new card in a few clicks!",
            },
            Object {
              "description": "Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.",
              "id": 3,
              "img": "onboarding3",
              "title": "Easy payments all over the world!",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={
          Array [
            Object {
              "onViewableItemsChanged": [Function],
              "viewabilityConfig": Object {
                "viewAreaCoveragePercentThreshold": 50,
              },
            },
          ]
        }
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "width": 650,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1B1D4D",
                    },
                    Object {
                      "fontFamily": "MulishBold",
                      "fontSize": 33,
                      "lineHeight": 36.3,
                    },
                    Array [
                      Object {
                        "marginVertical": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-heading-1-test-id"
              >
                Welcome to BankingBay app!
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4C4C60",
                      "fontFamily": "MulishRegular",
                      "fontSize": 17,
                      "lineHeight": 27.2,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 28.8,
                        "marginBottom": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-text-1-test-id"
              >
                Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "width": 650,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1B1D4D",
                    },
                    Object {
                      "fontFamily": "MulishBold",
                      "fontSize": 33,
                      "lineHeight": 36.3,
                    },
                    Array [
                      Object {
                        "marginVertical": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-heading-2-test-id"
              >
                Get a new card in a few clicks!
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4C4C60",
                      "fontFamily": "MulishRegular",
                      "fontSize": 17,
                      "lineHeight": 27.2,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 28.8,
                        "marginBottom": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-text-2-test-id"
              >
                Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "width": 650,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#1B1D4D",
                    },
                    Object {
                      "fontFamily": "MulishBold",
                      "fontSize": 33,
                      "lineHeight": 36.3,
                    },
                    Array [
                      Object {
                        "marginVertical": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-heading-3-test-id"
              >
                Easy payments all over the world!
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#4C4C60",
                      "fontFamily": "MulishRegular",
                      "fontSize": 17,
                      "lineHeight": 27.2,
                    },
                    Array [
                      Object {
                        "fontSize": 18,
                        "lineHeight": 28.8,
                        "marginBottom": 20,
                      },
                      Object {
                        "textAlign": "center",
                      },
                    ],
                  ]
                }
                testID="onboarding-text-3-test-id"
              >
                Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
        testID="onboarding-template-paginator-test-id"
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#1B1D4D",
                "borderRadius": 100,
                "borderWidth": 3,
                "height": 8,
                "marginHorizontal": 8,
                "opacity": 0.2,
                "width": 8,
              },
              Object {
                "opacity": 1,
              },
              Object {
                "borderColor": undefined,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#1B1D4D",
                "borderRadius": 100,
                "borderWidth": 3,
                "height": 8,
                "marginHorizontal": 8,
                "opacity": 0.2,
                "width": 8,
              },
              false,
              Object {
                "borderColor": undefined,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderColor": "#1B1D4D",
                "borderRadius": 100,
                "borderWidth": 3,
                "height": 8,
                "marginHorizontal": 8,
                "opacity": 0.2,
                "width": 8,
              },
              false,
              Object {
                "borderColor": undefined,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignSelf": "stretch",
            "marginBottom": 50,
            "marginTop": "auto",
          },
        ]
      }
      testID="onboarding-btn-test-id"
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288142079,
              4279654301,
            ],
            "endPoint": Array [
              1,
              1,
            ],
            "locations": undefined,
            "startPoint": Array [
              0,
              0,
            ],
          }
        }
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            }
          }
        >
          Get Started
        </Text>
      </ViewManagerAdapter_ExpoLinearGradient>
    </View>
  </View>
</View>
`;
